addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,498 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+import json
+
+from odoo import models, api, _
+from odoo.tools import float_round, float_repr
+from odoo.tools import html_escape
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = "account.edi.format"
+
+    def _is_enabled_by_default_on_journal(self, journal):
+        self.ensure_one()
+        if self.code == "in_einvoice_1_03":
+            return True
+        return super()._is_enabled_by_default_on_journal(journal)
+
+    def _is_required_for_invoice(self, invoice):
+        self.ensure_one()
+        if self.code == "in_einvoice_1_03" and invoice.is_sale_document():
+            return invoice.l10n_in_gst_treatment in (
+                "regular",
+                "composition",
+                "overseas",
+                "special_economic_zone",
+                "deemed_export",
+            )
+        return super()._is_required_for_invoice(invoice)
+
+    def _needs_web_services(self):
+        self.ensure_one()
+        return self.code == "in_einvoice_1_03" and True or super()._needs_web_services()
+
+    def _get_invoice_edi_content(self, move):
+        if self.code != "in_einvoice_1_03":
+            return super()._get_invoice_edi_content(move)
+        json_dump = json.dumps(self._l10n_in_edi_generate_json(move))
+        return json_dump.encode()
+
+    def _check_move_configuration(self, move):
+        if self.code != "in_einvoice_1_03":
+            return super()._check_move_configuration(move)
+        error_message = []
+        error_message += self._l10n_in_validate_partner(move.partner_id)
+        error_message += self._l10n_in_validate_partner(move.company_id.partner_id, is_company=True)
+        if not move.company_id.l10n_in_edi_username or not move.company_id.sudo().l10n_in_edi_password:
+            user_is_system = self.env._is_system()
```

Comment:
Okay, I remove it.

but I add it because If I user doesn't have a right then what is the meeting to tell him to set the user name and password in setting, so change msg based on the setting rights of the user.